src/wms/src/main/java/org/geoserver/wms/map/GetMapKvpRequestReader.java

```diff
@@ -288,13 +303,19 @@ public GetMapRequest read(Object request, Map kvp, Map rawKvp) throws Exception
         if (layerParam != null) {
             List<String> layerNames = KvpUtils.readFlat(layerParam);
             requestedLayerInfos.addAll(parseLayers(layerNames, remoteOwsUrl, remoteOwsType));
+        } else if (citeCompliant) {
+            throw new ServiceException(
+                    "GetMap LAYERS parameter is missing, it is mandatory for CITE compliant mode");
         }
 
         // raw styles parameter
         String stylesParam = (String) kvp.get("STYLES");
         List<String> styleNameList = new ArrayList<String>();
         if (stylesParam != null) {
             styleNameList.addAll(KvpUtils.readFlat(stylesParam));
+        } else if (citeCompliant) {
```

Comment:
Same as above for LAYERS, styles becomes non mandatory if SLD or SLD_BODY are specified.

Reply:
Thank you very much @aaime, I have changed the code and I have added more tests.